WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -816,17 +832,29 @@ extension OrderDetailsViewModel {
     /// Additionally it logs to tracks if the plugin store is accessed without it being in sync so we can handle that edge-case if it happens recurrently.
     /// Useful for when a plugin has had many names.
     ///
-    private func isPluginActive(_ plugins: [String], completion: @escaping (Bool) -> (Void)) {
+    private func isPluginActive(_ pluginNames: [String], completion: @escaping (Bool) -> (Void)) {
+        Task { @MainActor in
+            let plugin = await fetchPlugin(pluginNames)
+            completion(plugin?.active == true)
+        }
+    }
+
+    /// Fetches a plugin from storage, based on the provided list of plugin names.
+    /// Additionally it logs to tracks if the plugin store is accessed without it being in sync so we can handle that edge-case if it happens recurrently.
+    ///
+    @MainActor
+    private func fetchPlugin(_ pluginNames: [String]) async -> SystemPlugin? {
         guard arePluginsSynced() else {
-            DDLogError("⚠️ SystemPlugins acceded without being in sync.")
+            DDLogError("⚠️ SystemPlugins accessed without being in sync.")
             ServiceLocator.analytics.track(event: WooAnalyticsEvent.Orders.pluginsNotSyncedYet())
-            return completion(false)
+            return nil
         }
```

Comment:
I noticed that we are having quite a few occurrences of this event. We might want to update this approach to sync the plugins remotely first, before handling them here.